nn_combined.py

```diff
@@ -0,0 +1,95 @@
+from keras.models import Sequential
+from keras.layers.core import Activation, Dense, Dropout
+from keras.layers import Masking, GRU, Merge, Input, merge
+from keras.callbacks import EarlyStopping, ModelCheckpoint
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+import datetime
+import time
+
+def NN_train(data, model_name):
+    '''
+    Args:
+        data: dictionary containing relevant data
+     Returns:
+        recurrent neural network: A combined recurrent neural network trained on the different classes of the data
+    '''
+    
+    #defines training sets of different classes
+    X_jets_train=data['X_jet_train']
+    X_photons_train=data['X_photon_train']
+    y_train=data['y_train']
+
+    #set up sequential neural networks for the jet and photon classes 
+    jet_channel = Sequential()
+    photon_channel = Sequential()
+
+    #declaring the shape of the first row of each class matrix
+    JET_SHAPE = X_jets_train.shape[1:]
+    PHOTON_SHAPE = X_photons_train.shape[1:]
+
+    #adding layers to the jet and photon class neural networks
+    jet_channel.add(Masking(mask_value=-999, input_shape=JET_SHAPE, name='jet_masking'))
+    jet_channel.add(GRU(25, name='jet_gru'))
+    jet_channel.add(Dropout(0.3, name='jet_dropout'))
+
+    photon_channel.add(Masking(mask_value=-999, input_shape=PHOTON_SHAPE, name='photon_masking'))
+    photon_channel.add(GRU(10, name='photon_gru'))
+    photon_channel.add(Dropout(0.3, name='photon_dropout'))
+
+    #combining the jet and photon classes to make a combined recurrent neural network
+    combined_rnn = Sequential()
+    combined_rnn.add(Merge([jet_channel, photon_channel], mode='concat'))
+    combined_rnn.add(Dense(72, activation='relu'))
+    combined_rnn.add(Dropout(0.3))   
+    combined_rnn.add(Dense(36, activation='relu'))
+    combined_rnn.add(Dropout(0.3))
+    combined_rnn.add(Dense(24, activation='relu'))
+    combined_rnn.add(Dropout(0.3))
+    combined_rnn.add(Dense(12, activation='relu'))
+    combined_rnn.add(Dropout(0.3))
+    combined_rnn.add(Dense(6, activation='softmax'))
+
+    combined_rnn.compile('adam', 'sparse_categorical_crossentropy')
+
+    print 'Training:'
+    try:
+        combined_rnn.fit([X_jets_train, X_photons_train], 
+            y_train, batch_size=16, class_weight={
+                k : (float(len(y_train)) / float(len(np.unique(y_train)) * (len(y_train[y_train == k])))) for k in np.unique(y_train)
+            },
+            callbacks = [
+                EarlyStopping(verbose=True, patience=20, monitor='val_loss'),
+                ModelCheckpoint('./models/combinedrnn-progress',
```

Comment:
If you don't make the name of this file unique using your `model_name` or other methods, training different nets simultaneously is going to keep on overwriting this file.